File: 03.programmers/lv02/01.C++/09_biggest_num.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
bool compare(const string &a, const string &b) { return a + b > b + a; }
string solution(vector<int> numbers)
{
    string answer = "";
    vector<string> strArr;
    for (int i = 0; i < numbers.size(); i++)
        strArr.push_back(to_string(numbers.at(i)));

    sort(strArr.begin(), strArr.end(), compare);

    for (string str : strArr)
        answer += str;

    if (answer[0] == '0')
        return "0";
    return answer;
}
int main()
{
    vector<int> numbers = {3, 30, 34, 5, 9};
    cout << solution(numbers) << endl; 
    return 0;
}

File: ~2023/03.programmers/lv02/02.Python/09_biggest_num.py
def comp(a, b):
    return a+b > b+a


def solution(numbers):
    numbers.sort(key=comp)
    # numbers = sorted(numbers)
    print(numbers)


input_numbers = [3, 30, 34, 5, 9]
print(solution(input_numbers))
 